# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/chrome/common/child_thread.h
# Commit: 790fd8710dd7
# Full Hash: 790fd8710dd7bfa6085379e7f90b043e7aff383f
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 2
# Description:
#   Bug 1706374 - Part 12a: Initialize NodeController when creating IO thread, r=handyman
#   
#   This also consumes the existing channel created when launching a process to
#   create the the conneciton required by NodeController for communicating between
#   processes. In part 12b, consumers of the broken APIs will be adjusted to use
# ==============================================================================

diff -r 7b7819250f79 -r 790fd8710dd7 ipc/chromium/src/chrome/common/child_thread.h
--- a/ipc/chromium/src/chrome/common/child_thread.h	Tue Jun 22 18:17:22 2021 +0000
+++ b/ipc/chromium/src/chrome/common/child_thread.h	Tue Jun 22 18:17:22 2021 +0000
@@ -9,7 +9,9 @@
 
 #include "base/thread.h"
 #include "chrome/common/ipc_channel.h"
+#include "mojo/core/ports/port_ref.h"
 #include "mozilla/UniquePtr.h"
+#include "mozilla/ipc/ScopedPort.h"
 
 class ResourceDispatcher;
 
@@ -30,7 +32,9 @@
   // Returns the one child thread.
   static ChildThread* current();
 
-  mozilla::UniquePtr<IPC::Channel> TakeChannel() { return std::move(channel_); }
+  mozilla::ipc::ScopedPort TakeInitialPort() {
+    return std::move(initial_port_);
+  }
 
   // Thread implementation.
   virtual void Init() override;
@@ -45,7 +49,8 @@
   MessageLoop* owner_loop_;
 
   IPC::Channel::ChannelId channel_name_;
-  mozilla::UniquePtr<IPC::Channel> channel_;
+
+  mozilla::ipc::ScopedPort initial_port_;
 
   Thread::Options options_;
 